### PR TITLE
Check current user when deleting and updating organization

### PR DESCRIPTION
**Pull request by pulltheflower**
- Add current_user to EditOrgReq
- Add DeleteOrgReq
- Check permission of current_user when deleting and updating organiztion

### CODE REVIEW

File: api/handler/organization.go [hunk 1]
```diff
@@ -87,19 +87,27 @@ func (h *OrganizationHandler) Index(ctx *gin.Context) {
...
 // @Success      200  {object}  types.Response{} "OK"
 // @Failure      500  {object}  types.APIInternalServerError "Internal server error"
 // @Router       /organizations/{name} [delete]
 func (h *OrganizationHandler) Delete(ctx *gin.Context) {
-	name := ctx.Param("name")
-	err := h.c.Delete(ctx, name)
+	var req types.DeleteOrgReq
+	if err := ctx.ShouldBindJSON(&req); err != nil {
+		slog.Error("Bad request format", "error", err)
+		ctx.JSON(http.StatusBadRequest, gin.H{"message": err.Error()})
```
Comment (Rader) [inline api/handler/organization.go:98]: change to `httpbase.BadRequest`

File: common/types/organization.go [hunk 2]
```diff
@@ -15,7 +15,25 @@ type EditOrgReq struct {
 	Description string `json:"description" example:"org description"`
 	//TODO:rename json field name to 'name", need to negotiate with Portal engineer
 	//Org unique identifier
-	Name string `json:"path" example:"org_name_1"`
+	Name        string `json:"path" example:"org_name_1"`
+	CurrentUser string `json:"current_user"`
+}
+
+// This struct only used in swagger API
+type EditOrgBody struct {
```
Comment (Rader) [inline common/types/organization.go:23]: why this struct?

Comment (pulltheflower) [reply common/types/organization.go:23]: The swagger API of edit organization does not need parameter `name`, the `name` is already specified in the url. So I defined a new struct to represent the required fields in the API body.

File: api/handler/organization.go (hunk 1 above)
Comment (pulltheflower) [reply api/handler/organization.go]: Thanks for your review. Changed to use `httpbase.BadRequest`.

File: common/types/organization.go (hunk 2 above)
Comment (pulltheflower) [reply common/types/organization.go]: This problem is solved by using `json:"-"`.

**Review by Rader — APPROVED** [review]
LGTM

**Pull request merged**